apps/prairielearn/python/python_helper_sympy_test.py

```diff
@@ -196,7 +196,7 @@ def test_string_conversion_complex_conflict(self) -> None:
     def test_json_conversion(
         self,
         a_pair: tuple[str, sympy.Expr],
-        custom_functions: Optional[list[str]],
+        custom_functions: list[str] | None,
```

Comment:
👀 is `Optional` no longer recommended for use? What rule is actually making this change?

Reply:
https://docs.astral.sh/ruff/rules/non-pep604-annotation/

Reply:
I also prefer this syntax -- It's like `Optional<T>` in typescript vs `T | null`